ToolBox/KaikoDataConverter/KaikoDataConverterProgram.cs

```diff
@@ -13,42 +13,103 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
-using System;
-using System.Collections.Generic;
-using System.Globalization;
-using System.IO;
-using System.Linq;
+
 using NodaTime;
 using QuantConnect.Data;
 using QuantConnect.Data.Market;
 using QuantConnect.Logging;
 using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using ZipFile = Ionic.Zip.ZipFile;
 
 namespace QuantConnect.ToolBox.KaikoDataConverter
 {
     /// <summary>
-    /// Console application for converting Kaiko data into Lean data format
+    /// Console application for converting a single day of Kaiko data into Lean data format for high resolutions (tick, second and minute)
     /// </summary>
     public static class KaikoDataConverterProgram
     {
-        public static void KaikoDataConverter(string market, string _tickType, string sourceDirectory)
+        /// <summary>
+        /// Kaiko data converter entry point.
+        /// </summary>
+        /// <param name="sourceDirectory">The source directory where all Kaiko zipped files are stored..</param>
+        /// <param name="date">The date to process.</param>
+        /// <exception cref="ArgumentException">Source folder does not exists.</exception>
+        /// <remarks>This converter will process automatically data for every exchange and for both tick types if the raw data files are available in the sourceDirectory</remarks>
+        public static void KaikoDataConverter(string sourceDirectory, string date)
         {
-            if (market.IsNullOrEmpty() || _tickType.IsNullOrEmpty() || sourceDirectory.IsNullOrEmpty())
+            var timer = new Stopwatch();
+            timer.Start();
+            var folderPath = new DirectoryInfo(sourceDirectory);
+            if (!folderPath.Exists)
             {
-                throw new ArgumentException("The arguments must be specified as [market] [tickType (quote/trade)] [kaiko raw data directory path]");
+                throw new ArgumentException($"Source folder {folderPath.FullName} not found");
             }
-            var tickType = _tickType.ToLower() == "quote" ? TickType.Quote : TickType.Trade;
 
-            if (tickType == TickType.Quote)
-            {
-                CreateCryptoTicks(sourceDirectory, market, tickType, ParseKaikoQuoteFile);
-                AggregateTicksInAllResolutions(market, tickType);
-            }
-            else
+            foreach (var filePath in folderPath.EnumerateFiles("*.zip"))
             {
-                CreateCryptoTicks(sourceDirectory, market, tickType, ParseKaikoTradeFile);
-                AggregateTicksInAllResolutions(market, tickType);
+                var processingDate = DateTime.ParseExact(date, DateFormat.EightCharacter, CultureInfo.InvariantCulture);
```

Comment:
Maybe could move the `processingDate` above the for loop

Reply:
Done!